protocol/rpcprovider/rewardserver/reward_server.go

```diff
@@ -70,21 +70,22 @@ type RewardsTxSender interface {
 	EarliestBlockInMemory(ctx context.Context) (uint64, error)
 }
 
-func (rws *RewardServer) SendNewProof(ctx context.Context, proof *pairingtypes.RelaySession, epoch uint64, consumerAddr string) (existingCU uint64, updatedWithProof bool) {
+func (rws *RewardServer) SendNewProof(ctx context.Context, proof *pairingtypes.RelaySession, epoch uint64, consumerAddr string, apiInterface string) (existingCU uint64, updatedWithProof bool) {
 	rws.lock.Lock() // assuming 99% of the time we will need to write the new entry so there's no use in doing the read lock first to check stuff
 	defer rws.lock.Unlock()
+	consumerRewardsKey := proof.SpecId + apiInterface + consumerAddr
```

Comment:
this should be a function, that gets these as inputs and returns them unified that way no mistakes can happen in the order.

Reply:
fixed: 57dff5307fc3945fee74060280ef65ceaa19e079